### PR TITLE
change split to get ratio as input

### LINKED ISSUE #221: change split to get ratio as input
The current split uses the `splitAmount` to calculate. Instead, need to update it to a `ratio`.
https://github.com/The-Poolz/LockDealNFT/blob/18c1855bf3aa105fc5a5c1b180ba5e821429af12/contracts/LockDealNFT/LockDealNFT.sol#L105-L109

### PR DESCRIPTION
**Pull request by YouStillAlive**
closes #221

### CODE REVIEW

File: contracts/SimpleProviders/DealProvider/DealProvider.sol [hunk 1]
```diff
@@ -32,13 +35,14 @@ contract DealProvider is DealProviderModifiers, BasicProvider {
...
-        uint256 splitAmount
+        uint256 ratio
     )
         public
         override
         onlyProvider
-        invalidSplitAmount(poolIdToAmount[oldPoolId], splitAmount)
     {
+        uint256 splitAmount = poolIdToAmount[oldPoolId].calcAmount(ratio);
+        require(poolIdToAmount[oldPoolId] >= splitAmount, "Split amount exceeds the available amount");
```
Comment (Lomet) [inline contracts/SimpleProviders/DealProvider/DealProvider.sol:45]: if we check that the ratio can't be > 1e18 - how can we get here?

**Review by Lomet — APPROVED** [review]
Looks good

File: contracts/SimpleProviders/DealProvider/DealProvider.sol (hunk 1 above)
Comment (YouStillAlive) [reply contracts/SimpleProviders/DealProvider/DealProvider.sol:45]: If some other provider is using it.

Comment (YouStillAlive) [reply contracts/SimpleProviders/DealProvider/DealProvider.sol:45]: Miscalculation of the provider, which calls the split function.

**Review by Lomet — APPROVED** [review]

**Pull request merged**